### ISSUE TITLE
Intellisense errors due to extraneous include path elements passed to GCC on Windows 10

### ISSUE THREAD

**Issue by escreven** [microsoft/vscode-cpptools#1988]
I am running version 0.17.0 on Windows 10 with the Cygwin GCC compiler.  Opening a C++ source file always results in an Intellisense "#include" error, even when using settings files that worked without issue in a prior extension version.  

Using loggingLevel "6", I observe that at startup time the extension processes the "WINDOWS KITS" include file directories, and when a C++ file is opened, those same directories are sent to GCC as part of the include file path before the proper Cygwin include directories.  That leads to an error because there are conflicting includes files in the "WINDOWS KITS" directories ("string.h", for example).

This issue occurs whether c_cpp_properties includes a "Win32" configuration or not; explicitly defining and includePath or systemIncludePath doesn't seem to make a difference.

Version information:  
- C/C++ Extension 0.17.0
- VS Code - 1.23.1
- Windows - 10.0.16299 Build 16299
- GCC - 6.4.0

Attached are four files: a tiny C++ source file which causes an error when opened, the error message that pops up when hovering over the #include directive causing the error, the extension output when VS Code starts up, and the extension output when VS Code opens the C++ source file.

[tiny.cpp.txt](https://github.com/Microsoft/vscode-cpptools/files/1998084/tiny.cpp.txt)
[error-text.txt](https://github.com/Microsoft/vscode-cpptools/files/1998085/error-text.txt)
[output-startup.txt](https://github.com/Microsoft/vscode-cpptools/files/1998086/output-startup.txt)
[output-open.txt](https://github.com/Microsoft/vscode-cpptools/files/1998087/output-open.txt)

One more note: After upgrading to 0.17.0 from the prior version, opening C++ source files in existing directories previously opened using the prior extension worked fine.  Only source files in new directories, not previously opened using the older version failed.  Failures on new directories occurred even when I created new directories by copying the full contents of old ones (include .vscode).  Unfortunately, in an initial attempt to cure the problem, I fully deinstalled and reinstalled VS Code and the extension, effectively making all directories new -- so I can't provide diagnostic information related to the old directories.

**Comment by bobbrow** [comment]
Can you share your c_cpp_properties.json?  Or try setting `"compilerPath"` to something like `"c:/cygwin/bin/gcc.exe"` and `"intelliSenseMode"` to `"clang-x64"`

**Comment by escreven** [comment]
[Sorry for the delay in responding.  I didn't see your question immediately because of Yahoo Mail's overaggressive spam filter.]

While preparing different combinations of c_cpp_properities.json settings to try and then upload in answer to your question, I discovered the following: 
- My original c_cpp_properties.json settings file -- the one that worked with the prior extension version -- included both a "Win32" configuration and a "Win32-Original" configuration.  "Win32-Original" was the out of the box Win32 configuration which included "Win32 Kits" directories.  
- If I delete the "Win32-Original" configuration, 0.17.0 then processes the correct include path at startup and when opening a c++ source file.
- If I restore "Win32-Original" after successfully opening a folder once, 0.17.0 still functions correctly for c++ source files in that directory, including newly created ones.

Attached for reference is my original c_cpp_properties.json file.

[c_cpp_properties-ORIGINAL.json.txt](https://github.com/Microsoft/vscode-cpptools/files/1998890/c_cpp_properties-ORIGINAL.json.txt)

Thank you.

**Comment by escreven** [comment]
Examining the source, I see that getConfigIndexForPlatform in LanguageServer/configuration.ts returns the last configuration in c_cpp_properties if there are more than three configurations under the assumption (based on the comment) that this the user's desired custom configuration.  So, in my case, that would have returned the "Win32-Original" configuration, almost certainly causing the behavior I observed.

I missed this if documented, and it is surprising to me as a user, but now I know, and hopefully so does anyone else who might stumble onto this note!

Thank you.

**Comment by bobbrow** [comment]
Thanks for pointing this out.  That line should have been removed in 0.17.0, but we missed it.  I'll remove that logic now.

So did you open a folder that was preconfigured by someone else?  That method should only be called the first time you open a folder or if you make certain modifications to the configuration file.  From then on the extension is supposed to remember which config you were on.

**Comment by escreven** [comment]
No, all the folders I have used were created and configured by me.  

Now, it has been several weeks since I started using vscode and the extension, and the c_cpp_properties files in the pre-existing folders were in flux as I found my way around the tool.  It is possible that the configuration order was different early on, so the "Win32" (vs. "Win32-Original") configuration might have been last initially, allowing the older version of the extension to open that folder without issue.    But, I really don't remember.

In any case, everything works now.  Thank you for looking at this.
